api/src/main/java/com/spotify/ffwd/util/HighFrequencyDetector.java

```diff
@@ -170,13 +163,13 @@ private void updateHighFrequencyMetricsStats(final Map<Metric, Integer> groupedM
 
     String keys =
         highFrequencyMetrics.get().keySet().stream()
-            .sorted()
+            .sorted(Comparator.comparing(Metric::hashCode))
```

Comment:
I think in the previous PR you stop using Metric.HashCode as the metric identifier. Introducing it back here may bring some inconsistency.  Is having the  resulting keys order by hashcode relevant ? if not you can start by extracting the keys from the metric and work with that. Strings are comparable you don't have to provide a comparator.

Reply:
It is used for sorting only. It needs to be consistent when retrieving the tags (key and what).